tsconfig.build.json

```
@@ -0,0 +1,12 @@
+{
+  "extends": "./tsconfig.json",
+  "compilerOptions": {
+    "declaration": true,
+    "declarationDir": "dist",
+    "noEmit": false,
+    "outDir": "dist",
+    "rootDir": "src"
+  },
+  "include": ["./src/**/*"],
+  "exclude": ["./src/**/*.test.ts", "./src/**/*.test.js"]
```

There are no JS tests
```suggestion
  "exclude": ["./src/**/*.test.ts"]
```

---

Also, do we still need these two separate configs? I thought you reached the conclusion that one was sufficient. We still use just one in the template repo.

---

Hmm... that's true. Fair enough, I can remove this.

---

I am introducing a custom matcher to test that a promise never resolves, and in order to do this, I have to augment Jest's types. This seems to be only possible when using a TypeScript config that does *not* limit the known files using `include`. I have yet to know whether splitting the TypeScript config will be truly needed for all projects, so I have been hesitant in adding this to the module template, but at least for here I think it is valuable.